src/kv1/read.rs

```diff
@@ -195,32 +228,144 @@ pub fn make_physical_and_commercial_modes(collections: &mut Collections) {
+/// Read stop areas
+fn read_usrstar<H>(file_handler: &mut H) -> Result<BTreeMap<String, UsrStopArea>>
+where
+    for<'a> &'a mut H: FileHandler,
+{
+    let file = "USRSTARXXX.TMI";
+    let (file_reader, path) = file_handler.get_file(file)?;
+    info!("Reading {}", file);
 
-    Ok(())
+    let mut rdr = csv::ReaderBuilder::new()
+        .delimiter(b'|')
+        .trim(csv::Trim::All)
+        .from_reader(file_reader);
+    let usr_stop_area_map: BTreeMap<_, _> = rdr
+        .deserialize()
+        .into_iter()
+        .map(|usr_stop_area| usr_stop_area.with_context(ctx_from_path!(path)).unwrap())
+        .map(|usr_stop_area: UsrStopArea| (usr_stop_area.id.clone(), usr_stop_area))
+        .collect();
+    Ok(usr_stop_area_map)
 }
 
-/// Generates stop_areas
-pub fn read_usrstar<H>(file_handler: &mut H, collections: &mut Collections) -> Result<()>
+/// Generates stop_points
+pub fn read_usrstop_point<H>(file_handler: &mut H, collections: &mut Collections) -> Result<()>
 where
     for<'a> &'a mut H: FileHandler,
 {
-    info!("Reading USRSTARXXX.TMI");
-    // filter collections.stop_points by sp.parent == UserStopAreaCode to calculate barycenter
-    // collections.stop_areas = CollectionWithId::new(stop_areas)?;
+    let point_map = read_point(file_handler)?;
+    let mut usr_stop_area_map = read_usrstar(file_handler)?;
+
+    let file = "USRSTOPXXX.TMI";
+    let (file_reader, path) = file_handler.get_file(file)?;
+    info!("Reading {}", file);
+
+    let mut rdr = csv::ReaderBuilder::new()
+        .delimiter(b'|')
+        .trim(csv::Trim::All)
+        .from_reader(file_reader);
+
+    for usr_stop in rdr.deserialize() {
+        let usr_stop: UsrStop = usr_stop.with_context(ctx_from_path!(path))?;
+        let coord = match point_map.get(&usr_stop.point_code) {
+            Some(c) => c.clone(),
+            None => bail!("Point code {} does not exist.", usr_stop.point_code),
+        };
+        let stop_area_id = match usr_stop_area_map.get(&usr_stop.parent_station) {
+            Some(stop_area) => stop_area.id.clone(),
+            None => bail!(
+                "Stop Area with id {} does not exist.",
+                usr_stop.parent_station
+            ),
+        };
+        let stop_point = StopPoint {
+            id: usr_stop.point_code,
+            name: usr_stop.name,
+            codes: KeysValues::default(),
+            object_properties: KeysValues::default(),
+            comment_links: CommentLinksT::default(),
+            visible: true,
+            coord,
+            stop_area_id,
+            timezone: None,
+            geometry_id: None,
+            equipment_id: None,
+            fare_zone_id: None,
+            stop_type: StopType::Point,
+        };
+        collections.stop_points.push(stop_point).unwrap();
+    }
+
+    for (_, usr_stop_area) in usr_stop_area_map {
+        let stop_points = &collections.stop_points;
+        let stop_points: GeoMultiPoint<_> = stop_points
```

Comment:
```rust
let stop_points: GeoMultiPoint<_> = collections
            .stop_points
            .values()
            .filter(|sp| sp.stop_area_id == usr_stop_area.id)
            .map(|sp| (sp.coord.lon, sp.coord.lat))
            .collect();

        let coord = stop_points.centroid().map(|c| Coord {
                lon: c.x(),
                lat: c.y(),
            }).ok_or_else(|| format_err!(
                "Failed to calculate a barycenter of stop area {} because it doesn't refer to any corresponding stop point.",
                usr_stop_area.id
            ))?;
```

Reply:
Fixed in b0f23e74271ab57c3b906a822bcd9951415bbb57.